### PR TITLE
[SPEC-7510] Remove unnecessary static container for AWSCore scriptcanvas nodes

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
Remove unnecessary static vector migrated from LY1.X

## Testing
Pass unit tests
Automated testing pass

### CODE REVIEW

**Review by amznestebanpapp — APPROVED** [review]

File: Gems/AWSCore/Code/Source/ScriptCanvas/AWSScriptBehaviorsComponent.cpp [hunk 1]
```diff
@@ -12,35 +12,17 @@
...
-            m_behaviors.push_back(AZStd::make_unique<AWSScriptBehaviorLambda>());
-            m_behaviors.push_back(AZStd::make_unique<AWSScriptBehaviorS3>());
-            m_alreadyAddedBehaviors = true;
-        }
-    }
-
     void AWSScriptBehaviorsComponent::Reflect(AZ::ReflectContext* context)
     {
-        AddBehaviors();
+        AWSScriptBehaviorDynamoDB::Reflect(context);
```
Comment (lmbr-pip) [inline Gems/AWSCore/Code/Source/ScriptCanvas/AWSScriptBehaviorsComponent.cpp:17]: Please add:
* Comment that new script behaviours need to be added to list below as you are no longer dynamically adding them
* Docs story to update how developers add AWS Script Behaviours.

**Review by lmbr-pip — COMMENTED** [review]
LG, mostly

Can you confirm you ran script canvas / automation tests? Need to confirm the nodes are still exposed as expected. Testing section only mentions you ran unit tests.

File: Gems/AWSCore/Code/Source/ScriptCanvas/AWSScriptBehaviorsComponent.cpp (hunk 1 above)
Comment (onecent1101) [reply Gems/AWSCore/Code/Source/ScriptCanvas/AWSScriptBehaviorsComponent.cpp:17]: no comments needed, as originally it is not dynamically added.
no doc needed, as it is not specific to AWS, it is general way to create script canvas node, developer should have basic scripting knowledge to do it.

**Comment by onecent1101** [comment]
> 
> 
> LG, mostly
> 
> Can you confirm you ran script canvas / automation tests? Need to confirm the nodes are still exposed as expected. Testing section only mentions you ran unit tests.

Automated testing pass, updated PR details

**Review by junbo75 — APPROVED** [review]

**Pull request merged**